services:
    aetheryte-nginx:
        build:
            context: nginx
            dockerfile: Dockerfile
        networks:
            transcendence :
                ipv4_address: 172.20.0.4
        env_file:
            - ../../.env
        restart: on-failure

    aetheryte-postgres:
        build:
            context: postgres
            dockerfile: Dockerfile
        volumes:
            - aetheryte_postgres:/var/lib/postgresql/data/
        networks:
            transcendence:
                ipv4_address: 172.20.0.3
        environment:
            POSTGRES_DB: ${AETHERYTE_POSTGRES_DB}
            POSTGRES_USER: ${AETHERYTE_POSTGRES_USER}
            POSTGRES_PASSWORD: ${AETHERYTE_POSTGRES_PASSWORD}
        healthcheck:
            test: ["CMD-SHELL", "sh -c 'pg_isready -U ${AETHERYTE_POSTGRES_USER} -d ${AETHERYTE_POSTGRES_DB}'"]
            interval: 10s
            timeout: 5s
            retries: 10
        restart: unless-stopped

    aetheryte-python:
        build:
            context: python
            dockerfile: Dockerfile
        volumes:
            - ../app:/app
        networks:
            transcendence :
                ipv4_address: 172.20.0.2
        env_file:
            - ../../.env
        healthcheck:
            test: curl --fail http://localhost:8000/api/user || exit 1
            interval: 10s
            timeout: 5s
            retries: 10
        restart: on-failure